gammapy/modeling/models/spectral.py

```diff
@@ -316,13 +319,35 @@ def plot(
 
         energy_min, energy_max = energy_range
         energy = MapAxis.from_energy_bounds(
-            energy_min, energy_max, n_points, energy_unit
-        ).edges
+                                            energy_min, energy_max, n_points, energy_unit
+                                            ).edges
+
+        if sed_type:
+            if sed_type == "dnde":
+                energy_power = 0
+                flux = self(energy).to(flux_unit)
+
+            if sed_type == "e2dnde":
+                energy_power = 2
+                flux = self(energy).to(flux_unit)
 
-        # evaluate model
-        flux = self(energy).to(flux_unit)
+            if sed_type == "flux":
+                energy_power = 0
+                flux = self.integral(energy[:-1], energy[1:]).to("cm-2 s-1")
+                energy = (energy[:-1] + energy[1:]) / 2.
+
+            if sed_type == "eflux":
+                energy_power = 0
+                flux = self.energy_flux(energy[:-1], energy[1:]).to("TeV cm-2 s-1")
+                energy = (energy[:-1] + energy[1:]) / 2.
+
+            y = self._plot_scale_flux(energy, flux, energy_power)
+
+        else:
```

Comment:
Once you change to the proposed patter above, this else block shouldn't be needed....

Reply:
OK.